sympy/core/exprtools.py

```diff
@@ -1414,156 +1288,614 @@ def factor_nc(expr):
+        of degree ldegree and self.degree - ldegree, if possible.
+
+        Return (None, None) if no such factorization exists.
+
+        Implementation based on Algorithm 1 from https://arxiv.org/abs/1002.3180
+        """
+        terms = list(self.monomials(self.degree))
+        Ghat, Hhat = terms[0].split(ldegree)
+        G = _NCPoly(None, self.generators, self.extensions)
+        H = _NCPoly(None, self.generators, self.extensions)
+        G.add(Ghat)
+        H.add(Hhat)
+
+        for monomial in terms[1:]:
+            R = monomial.ldivide(Ghat)
+            if R is None:
+                L = monomial.rdivide(Hhat)
+                if L is not None:
+                    G.add(L)
             else:
-                n = common_suffix(n, a[1])
-            if not n:
-                # is there a power that can be extracted?
-                if not args[0][1]:
-                    break
-                b, e = args[0][1][-1].as_base_exp()
-                ok = False
-                if e.is_Integer:
-                    for t in args:
-                        if not t[1]:
-                            break
-                        bt, et = t[1][-1].as_base_exp()
-                        if et.is_Integer and bt == b:
-                            e = min(e, et)
-                        else:
-                            break
-                    else:
-                        ok = hit = True
-                        r = b**e
-                        il = b**-e
-                        for _ in args:
-                            _[1][-1] = _[1][-1]*il
-                        break
-                if not ok:
+                H.add(R)
+        if len(G) * len(H) != len(terms):
+            return None, None
+        GH = G * H
+        if GH.rep[self.degree] == self.rep[self.degree]:
+            return G, H
+        return None, None
+
+
+    def _factor_once(self, ldegree):
+        """Factor into pairs of polynomials of degree ldegree and self.degree - ldegree, if possible.
+        Coefficients will not be factored or otherwise simplified.
+
+        Return a generator that yields tuples of polynomials.
+
+        Implementation based on Algorithm 2 from https://arxiv.org/abs/1002.3180
+        """
+        from sympy.solvers.polysys import solve_poly_system
+        if self.degree == 0:
+            return
+        if len(self) < 2:
+            return
+
+        if ldegree == 0 or ldegree >= self.degree:
+            return
+
+        G, H = self._factor_homogeneous(ldegree)
+        if G is None:
+            return
+
+        n = self.degree
+        h = ldegree
+        k = n - ldegree
+        # Future optimization: select the monomials with the least overlap potential
+        Ghat = next(G.monomials())
+        Hhat = next(H.monomials())
+
+        dummies = []
+        Fhat = self.copy()
+
+        for j in range(1, n + 1):
+            for i in range(1, j):
+                Fhat -= G.copy(h - i) * H.copy(k - j + i)
```

Comment:
I think that you should have second experiment to get rid of `.copy`. I see you have already implemented `__add__` and `__mul__` with defensive copying, so this could be redundant